runtime/kusama/src/lib.rs

```diff
@@ -1326,6 +1327,97 @@ impl pallet_nomination_pools::Config for Runtime {
 	type MaxPointsToBalance = MaxPointsToBalance;
 }
 
+pub struct RemoveGovV1Storage;
+impl RemoveGovV1Storage {
+	fn get_pallets() -> Vec<&'static str> {
+		vec![
+			"Democracy",
+			"Council",
+			"TechnicalCommittee",
+			"PhragmenElection",
+			"TechnicalMembership",
+			"Treasury",
+		]
+	}
+
+	fn count_keys(prefix: &[u8]) -> u32 {
+		let mut current = prefix.clone().to_vec();
+		let mut counter = 0;
+		while let Some(next) = sp_io::storage::next_key(&current[..]) {
+			if !next.starts_with(prefix) {
+				return counter
+			}
+			counter += 1;
+			current = next;
+		}
+		0
+	}
+}
+impl frame_support::traits::OnRuntimeUpgrade for RemoveGovV1Storage {
+	fn on_runtime_upgrade() -> frame_support::weights::Weight {
+		use frame_support::dispatch::GetDispatchInfo;
+		use sp_runtime::traits::Dispatchable;
+
+		// Create a 'kill_prefix' call for each pallet we wish to remove keys for
+		let mut total_keys = 0;
+		let calls = Self::get_pallets()
+			.iter()
+			.map(|pallet| {
+				let prefix = twox_128(pallet.as_bytes());
+				let subkeys = Self::count_keys(&prefix);
+				total_keys += subkeys;
+				RuntimeCall::System(frame_system::Call::<Runtime>::kill_prefix {
+					prefix: prefix.to_vec(),
+					subkeys,
+				})
+			})
+			.collect::<Vec<_>>();
+
+		if total_keys > 0 {
+			// Wrap all 'kill_prefix' calls in a single batch transaction
+			log::info!(target: "runtime::kusama", "removing {} Gov V1 keys 💣", total_keys);
+			let batch = pallet_utility::Call::<Runtime>::batch { calls };
+			let estimated_weight = batch.get_dispatch_info().weight;
+			match RuntimeCall::Utility(batch).dispatch(RuntimeOrigin::root()) {
+				Ok(post_info) => post_info.actual_weight.unwrap_or_default(),
+				Err(_) => estimated_weight,
```

Comment:
I wasn't exactly sure what to return for weight in this arm. Let me know if there's something else I should do.